pkg/ext/utils.go

```diff
@@ -138,7 +138,10 @@ func CreateOrUpdate[T runtime.Object](
 			return nil, false, convertError(fmt.Errorf("object was of type %T, not of expected type %T", obj, zeroT))
 		}
 
-		newObj, err := createFn(ctx, tObj, &metav1.CreateOptions{})
+		newObj, err := createFn(ctx, tObj, &metav1.CreateOptions{
+			DryRun:       options.DryRun,
+			FieldManager: options.FieldManager,
```

Comment:
Do we need to include the FieldValidation option as well here?

Reply:
We do 😄 https://github.com/rancher/steve/pull/623#discussion_r2073784845